tests/tests.rs

```diff
@@ -211,3 +370,35 @@ fn test_provider_registry() {
         ]
     )
 }
+
+#[test]
+fn free_rpc_auth() {
+    let setup = EvmRpcSetup::new();
+    setup.authorize_caller(Auth::FreeRpc);
+
+    let url = "https://cloudflare-eth.com";
+    let payload = r#"{"id":1,"jsonrpc":"2.0","method":"eth_gasPrice","params":null}"#;
+    let expected_result = r#"{"id":1,"jsonrpc":"2.0","result":"0x00112233"}"#;
+    let result = setup
+        .request(
+            Source::Custom {
+                url: url.to_string(),
+                headers: None,
+            },
+            payload,
+            1000,
+        )
+        .mock_http(
+            MockOutcallBuilder::new(200, expected_result)
+                .expect_url(url.to_string())
+                .expect_method(HttpMethod::GET)
```

Comment:
@rvanasa : This I don't understand, it should be a `POST` request and not a `GET` request. If the test passes, then it also means that either there is a bug in the productive logic or there is a bug in the test infrastructure controlling the method.

Reply:
That is a good point; thanks for catching this. Adding some more tests to confirm that the mock HTTP logic works as expected.